### ISSUE TITLE
bug: vcs init container is using git credentials even when they aren't defined

### ISSUE THREAD

**Issue by krancour** [brigadecore/brigade#1225]
I believe this will clear up naturally when #1134 is solved, but just in case, I don't want to lose track of it.

The VCS init container isn't being set up to conditionally use git credentials only when they've been specified. It's unconditionally using them, and that's causing an issue like this:

```
Events:
  Type     Reason     Age                  From                         Message
  ----     ------     ----                 ----                         -------
  Normal   Scheduled  31m                  default-scheduler            Successfully assigned brigade-github-demo-fcd3d063-0595-4e2f-93d2-299e1bc404ff/eb5e7e93-6db4-428f-b3da-00cedc7b5770 to kind-control-plane
  Normal   Pulled     30m (x8 over 31m)    kubelet, kind-control-plane  Successfully pulled image "brigadecore/git-sidecar:v1.4.0"
  Warning  Failed     30m (x8 over 31m)    kubelet, kind-control-plane  Error: couldn't find key gitSSHKey in Secret brigade-github-demo-fcd3d063-0595-4e2f-93d2-299e1bc404ff/eb5e7e93-6db4-428f-b3da-00cedc7b5770
  Normal   Pulling    81s (x133 over 31m)  kubelet, kind-control-plane  Pulling image "brigadecore/git-sidecar:v1.4.0"
```

**Comment by vdice** [comment]
Similar to https://github.com/brigadecore/brigade/issues/1224, this should be fixed (and vetted by the tests therein) by https://github.com/brigadecore/brigade/pull/1229

**Comment by vdice** [comment]
We're not seeing this now that we have the v2 git initializer up and running.  Can we close @krancour ?

**Comment by krancour** [comment]
Yes. This was fixed with your excellent work on #1229
